src/chocolatey/infrastructure.app/services/NugetService.cs

```diff
@@ -862,8 +862,19 @@ public void remove_rollback_directory_if_exists(string packageName)
 
 			foreach (var packageName in packageNames)
 			{
-				// LastOrDefault to get the latest package from local repository, because all packages have set IsLatestVersion to true
-				var latestPackage = packageManager.SourceRepository.GetPackages().Where(x => x.Id.ToLower() == packageName && x.IsLatestVersion).LastOrDefault();
+				// If using local repository as source all packages have set IsLatestVersion to true.
+				var latestPackages = packageManager.SourceRepository.GetPackages().Where(x => x.Id.ToLower() == packageName && x.IsLatestVersion).ToList();
+
+				IPackage latestPackage = null;
+
+				if(latestPackages.Count > 1)
+				{
+					latestPackage = latestPackages.Last();

```

Comment:
There is actually a way to see if you are hitting a file repo or an OData repo. I'd take a look at that as this is not quite right.

Reply:
What do you suggest?

Reply:
Apologies meant to link it. Take a look at https://github.com/chocolatey/choco/blob/cdf0938f43bf6f2ac52d7ec5d27e4746f0d16436/src/chocolatey/infrastructure.app/nuget/NugetList.cs#L40

Reply:
Particularly https://github.com/chocolatey/choco/blob/cdf0938f43bf6f2ac52d7ec5d27e4746f0d16436/src/chocolatey/infrastructure.app/nuget/NugetList.cs#L50-L59